2765. Longest Alternating Subarray/solution.py

```diff
@@ -0,0 +1,68 @@
+class Solution:
+    # Given a array of integers, nums, find the longest alternating
+    # subarray within nums. A subarray is a contiguous, non-empty 
+    # sequence of elements in nums that meet the following criteria:
+    #   - The subarray is longer than 1 element
+    #   - The subarray starts with an integer n and the next element
+    #     must be n+1
+    #   - From then on, the difference between each successive element
+    #     of the subarray alternates between -1 and 1.
+    # Return -1 if there is no such subarray
+    def alternatingSubarray(self, nums: List[int]) -> int:
+        # Track max and current subarray length
+        maxLength = 0
+        curLength = 0
+
+        # Track whether the difference between successive elements
+        # should be 1 or -1. In this problem, the subarray must start
+        # with a difference of 1
+        expectedDiff = 1
+
+        # Iterate through nums to look for subarrays. Use index
+        # based iteration so we can look at two elements.
+        i = 0
+        while i < len(nums) - 1:
+
+            # Check if the current element and next element have a 
+            # difference that meets the conditions. 
+            if nums[i+1] - nums[i] != expectedDiff:
+                # They don't meet the conditions. Check if we need
+                # to update max subarray length before reseting the
+                # current length counter.
+                maxLength = max(maxLength, curLength)
```

Comment:
Actually, you only need to write 1 more line of code to achieve the same functionality with a lesser runtime:
if (curLength > maxLength):
  maxLength = curLength

That way, maxLength is only updated if curLength is greater than maxLength. You don't need to update maxLength if maxLength > curLength.